freqtrade/data/history/hdf5datahandler.py

```diff
@@ -28,13 +28,22 @@ def ohlcv_get_available_data(cls, datadir: Path) -> ListPairsWithTimeframes:
         :param datadir: Directory to search for ohlcv files
         :return: List of Tuples of (pair, timeframe)
         """
-        _tmp = [re.search(r'^([a-zA-Z_]+)\-(\d+\S+)(?=.h5)', p.name)
-                for p in datadir.glob("*.h5")]
-        return [(match[1].replace('_', '/'), match[2]) for match in _tmp
+        _tmp = [
+            re.search(
+                r'^([a-zA-Z_]+)\-(\d+\S)\-?([a-zA-Z_]*)?(?=.h5)',
+                p.name
+            ) for p in datadir.glob("*.h5")
+        ]
+        return [(match[1].replace('_', '/'), match[2], match[3]) for match in _tmp
```

Comment:
We will need to think for a solution to this.
right now, we assume to replace all `"_"` with `"/"`.
this works for spot-pairs - but is wrong for (future) futures pairs (i knew this for a while - and this "should" be the only place left with "/" assuming to split the pair)

If you look at the function in misc (`pair_to_filename()`) - we replace all special chars to "_".
this means, `BTC/USDT:USDT` will become `BTC_USDT_USDT` - which would translate back to `BTC/USDT/USDT` (now it's wrong, no longer matching the pairname).

Most likely, our approach will need to involve some ... 
more clever logic - maybe replacing ":" with something else to allow for "back-replacement".

Reply:
We could just use a better regex

Reply:
can/could also work - i'm just pointing out where i think there'll be problem for now - i don't "really" got solutions at this point :laughing:

Reply:
Unless windows doesn't `:` or `-` in file names, I don't think we need to change it

I updated the regex to detect `swap` and `futures` symbols

Reply:
i don't think `:` is valid as character for [windows](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions), as that's what's used to specify drive letters (`C:/xxx`).

`-` is for sure valid - i'm i think we're replacing that with `_` for "reasons" - and not for something for real ...

Reply:
Ok, I wrote the regex to accept both Perpetual Swap(ADA/USDT:USDT) and Delivery Futures(ADA/USDT:USDT-20220910) symbols, just because this is kind of the only part of this PR that would need accommodation for Delivery Futures, and it took me an extra couple minutes to implement the logic and save some future dev a lot of time going through these files

Reply:
That regex is now no longer working (at least not for json data).
simply running `freqtrade list-data --exchange binance` on a user-data directory with 2 pairs in yields only an exception - because there is no longer a 2nd match.